packages/teleport/src/DesktopSession/TdpClientCanvas/useTdpClientCanvas.tsx

```diff
@@ -32,21 +32,22 @@ export default function useTdpClientCanvas(props: Props) {
 
   // Build a client based on url parameters.
   const tdpClient = useMemo(() => {
+    const { width, height } = getDisplaySize();
+
     const addr = cfg.api.desktopWsAddr
       .replace(':fqdm', getHostName())
       .replace(':clusterId', clusterId)
       .replace(':desktopName', desktopName)
-      .replace(':token', getAccessToken());
+      .replace(':token', getAccessToken())
+      .replace(':username', username)
+      .replace(':width', width.toString())
+      .replace(':height', height.toString());
 
     return new TdpClient(addr, username);
   }, [clusterId, username, desktopName]);
 
-  const syncCanvasSizeToClientSize = (canvas: HTMLCanvasElement) => {
-    // Calculate the size of the canvas to be displayed.
-    // Setting flex to "1" ensures the canvas will fill out the area available to it,
-    // which we calculate based on the window dimensions and TopBarHeight below.
-    const width = window.innerWidth;
-    const height = window.innerHeight - TopBarHeight;
+  const syncCanvasSizeToDisplaySize = (canvas: HTMLCanvasElement) => {
+    const { width, height } = getDisplaySize();
 
     // If it's resolution does not match change it
     if (canvas.width !== width || canvas.height !== height) {
```

Comment:
Is it needed `if (canvas.width !== width || canvas.height !== height) {` ? Why not always assigning these to canvas?

Reply:
I can't recall, I think it was copied from a tutorial many moons ago. Doesn't seem necessary.